src/AutoUpdate/TranslationStatus.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace WPML\PB\AutoUpdate;
+
+use Exception;
+use WPML_Post_Element;
+use function WPML\Container\make;
+
+class TranslationStatus {
+
+	/**
+	 * @param WPML_Post_Element $element
+	 *
+	 * @return int|null
+	 */
+	public static function get( WPML_Post_Element $element ) {
+		try {
+			return make( '\WPML_TM_Translation_Status' )
+				->filter_translation_status( null, $element->get_trid(), $element->get_language_code() );
+		} catch ( Exception $e ) {}
+	}
```

Comment:
@strategio I'm not sure what will make it throw an exception but I assume it's `WPML_TM_Translation_Status` not being available.

I'm using this pattern for this:
```php
return Maybe::fromNullable( make( '\WPML_TM_Translation_Status' ) )
		->map( invoke( 'filter_translation_status' )->with( null, $element->get_trid(), $element->get_language_code() ) )
		->getOrElse( null );
```

Reply:
@strategio Notice this class has nothing to do with auto update except for being a helper. Ideally we should move this type of thing into a library. It would have to go into the core-api library so it's available even if TM is not available.

Reply:
Yes, that's the reason @brucepearson. We had one similar issue in the past so I decided to use this small wrapper class.

I also wanted to introduce this in the core-api library, but I was unsure as I don't have enough background with it. For instance, I decided to pass it a `\WPML_Post_Element` because it's convenient for me here, but I am not sure I would do that way in the library. I would rather try to use a convenient signature for what we already have in `WPML\Element\API\PostTranslations`.

I will try in a separate task (as a training case).